src/sql/base/browser/ui/chart/chart.component.ts

```diff
@@ -0,0 +1,321 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+import { Component, Inject, forwardRef, ChangeDetectorRef } from '@angular/core';
+import * as chartjs from 'chart.js';
+import { mixin } from 'sql/base/common/objects';
+import { Disposable } from 'vs/base/common/lifecycle';
+import * as azdata from 'azdata';
+
+export interface BarDataSet {
+	label: string;
+	data: number[];
+	backgroundColor?: string[];
+	borderColor?: string[];
+}
+
+@Component({
+	selector: 'chart-component',
+	templateUrl: decodeURI(require.toUrl('./chart.component.html'))
+})
+export class Chart<TConfig extends azdata.ChartConfiguration> extends Disposable {
+	private _chartId: string;
+	private _type: azdata.ChartType;
+	private _configuration: chartjs.ChartData;
+
+	public chart: chartjs.Chart;
+	private canvas: HTMLCanvasElement;
+	public element: string;
+
+	/**
+	 * Options in the form that Chart.js accepts (hence the `any` type)
```

Comment:
Why? We should be using the typings from them, not hoping we got them right (which can go badly if/when we update the library if they change stuff)

Reply:
Artifact from earlier prototyping; fixed